tap_appstore/client.py

```diff
@@ -18,13 +23,138 @@
 
 _Auth = Callable[[requests.PreparedRequest], requests.PreparedRequest]
 
-# TODO: Delete this is if not using json files for schema definition
-SCHEMAS_DIR = importlib_resources.files(__package__) / "schemas"
-
+logger = logging.getLogger(__name__)
 
 class AppStoreStream(RESTStream):
     """AppStore stream class."""
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.api = self.setup_api_connection()
+        self.float_fields = None
+        self.int_fields = None
+        self.date_fields = None
+
+    def setup_api_connection(self):
+        """Set up the API connection using provided configuration."""
+        return Api(self.config['key_id'], self.config['key_file'], self.config['issuer_id'])
+
+    def get_start_date(self, date_format='%Y-%m-%d', default_date='2024-01-01'):
+        """Retrieve the configured start date, formatted as specified."""
+        start_date_str = self.config.get('start_date', default_date)
+        try:
+            return datetime.strptime(start_date_str, date_format)
+        except ValueError as e:
+            logger.error(f"Invalid start date format: {start_date_str}. Error: {e}")
+            return None
+
+    def get_report_date(self, date):
+        """Return the report date formatted according to the specific needs of the stream."""
+        return date.strftime('%Y-%m-%d')
+
+    def download_data(self, start_date, api):
+        """Set up the endpoint for the API call. Override in subclass as needed."""
+        return None
+
+    def process_record(self, record):
+        """Convert ints, floats, and date fields"""
+        if self.float_fields:
+            self.convert_fields(record, self.float_fields, float)
+
+        if self.int_fields:
+            self.convert_fields(record, self.int_fields, int)
+
+        if self.date_fields:
+            for field, format in self.date_fields.items():
+                if field in record and record[field]:
+                    record[field] = self.convert_date(record[field], format)
```

Comment:
I think singer sdk tries to convert the field to the format specified in the schema automatically. It will raise an error if the conversion is no possible like trying to convert "a" to an int.

Reply:
I couldn't seem to get this to work. Will look into this more in a pairing session with you.